packages/web3-eth/src/utils/format_transaction.ts

```diff
@@ -17,18 +17,23 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 
 import { Transaction, DataFormat, DEFAULT_RETURN_FORMAT, FormatType } from 'web3-types';
 import { isNullish, ValidationSchemaInput } from 'web3-validator';
+import { mergeDeep, format, bytesToHex, toHex } from 'web3-utils';
 import { TransactionDataAndInputError } from 'web3-errors';
-import { mergeDeep, bytesToHex, format } from 'web3-utils';
-import { transactionSchema } from '../schemas.js';
+
+import { transactionInfoSchema, transactionSchema } from '../schemas.js';
 
 export function formatTransaction<
 	ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT,
 	TransactionType extends Transaction = Transaction,
 >(
 	transaction: TransactionType,
 	returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
-	options: { transactionSchema: ValidationSchemaInput | typeof transactionSchema } = {
-		transactionSchema,
+	options: {
+		transactionSchema?: ValidationSchemaInput | typeof transactionSchema;
+		fillInputAndData?: boolean;
```

Comment:
A new parameter, `fillInputAndData`, was added to the `options` argument for `formatTransaction`. This method is used for formatting both user submitted and RPC response transaction objects. So, we need a flag to determine whether to leave the transaction object as given, or fill in the `input` or `data` properties if they weren't present

Reply:
This, and things similar, needs to be in the documentation.